### PR TITLE
bot fixes

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9179

## Description
Jira tickets reopen for test modeling rules.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/scripts/jira_issues.py [hunk 1]
```diff
@@ -62,45 +62,51 @@ def jira_color_text(text: str, color: str) -> str:
...
-
-                # Find the transition with the specified ID
-                unresolved_transition = None
-                for transition in transitions:
-                    if transition['name'] == JIRA_ISSUE_UNRESOLVED_TRANSITION_NAME:
-                        unresolved_transition = transition
-                        break
-                if unresolved_transition:
-                    jira_server.transition_issue(searched_issue, unresolved_transition['id'])
+                unresolved_transition_id = find_jira_ticket_unresolved_transition(jira_server, jira_issue)
```
Comment (kobymeir) [inline Tests/scripts/jira_issues.py:96]: why no call this immediately before calling `transition_jira_ticket_to_unresolved` 
you're just passing the return value here for no good reason.

Comment (kobymeir) [inline Tests/scripts/jira_issues.py:96]: why no call this immediately before calling `transition_jira_ticket_to_unresolved` 
you're just passing the return value here for no good reason.

File: Tests/scripts/jira_issues.py [hunk 2]
```diff
@@ -62,45 +62,51 @@ def jira_color_text(text: str, color: str) -> str:
...
+            unresolved_transition = transition
+            break
+    return unresolved_transition['id'] if unresolved_transition else None
+
+
+def transition_jira_ticket_to_unresolved(jira_server: JIRA, jira_issue: Issue | None, unresolved_transition_id: str | None):
+    if unresolved_transition_id:
+        jira_server.transition_issue(jira_issue, unresolved_transition_id)
+
+
```
Comment (kobymeir) [inline Tests/scripts/jira_issues.py:80]: why did you split into 2 functions? honestly i don't see the point.

File: Tests/scripts/jira_issues.py [hunk 3]
```diff
@@ -62,45 +62,51 @@ def jira_color_text(text: str, color: str) -> str:
...
 
+def find_jira_ticket_unresolved_transition(jira_server: JIRA, jira_issue: Issue | None):
+    unresolved_transition = None
+    #  Get the available transitions for the issue
+    transitions = jira_server.transitions(jira_issue)
+    for transition in transitions:
+        if transition['name'] == JIRA_ISSUE_UNRESOLVED_TRANSITION_NAME:
+            unresolved_transition = transition
+            break
+    return unresolved_transition['id'] if unresolved_transition else None
```
Comment (kobymeir) [inline Tests/scripts/jira_issues.py:73]: i know i wrote this initially.
but it can be simplified into a single line with `next(filter(...`

**Review by kobymeir — CHANGES_REQUESTED** [review]

File: Tests/scripts/jira_issues.py (hunk 1 above)
Comment (sapirshuker) [reply Tests/scripts/jira_issues.py:96]: This allows me to make only one call to the jira_server.transitions function and to
keep the old functionallty that checks if there is a transition_id

File: Tests/scripts/jira_issues.py (hunk 2 above)
Comment (sapirshuker) [reply Tests/scripts/jira_issues.py:80]: This allows me to make only one call to the jira_server.transitions function

**Review by kobymeir — APPROVED** [review]

**Pull request merged**